services/event_service.go

```diff
@@ -151,7 +152,36 @@ func (e *EventService) GetAppEvent(ctx context.Context, id string) (*datastore.E
 
 func (e *EventService) Search(ctx context.Context, filter *datastore.Filter) ([]datastore.Event, datastore.PaginationData, error) {
 	var events []datastore.Event
-	ids, paginationData, err := e.searcher.Search("events", filter)
+	filterByBuilder := new(strings.Builder)
+
+	filterByBuilder.WriteString(fmt.Sprintf("group_id:=%s", filter.Group.UID))
+
+	hasAppFilter := !util.IsStringEmpty(filter.AppID)
+	if hasAppFilter {
+		filterByBuilder.WriteString(fmt.Sprintf(" && app_id:=%s", filter.Group.UID))
+	}
+
+	filterByBuilder.WriteString(fmt.Sprintf(" && created_at:[%d..%d]", filter.SearchParams.CreatedAtStart, filter.SearchParams.CreatedAtEnd))
+
+	documents, paginationData, err := e.searcher.Search(filter.Group.UID, &datastore.SearchFilter{
+		Query:    filter.Query,
+		FilterBy: filterByBuilder.String(),
+		Pageable: filter.Pageable,
+	})
+
+	var ids []string
+	for _, d := range documents {
```

Comment:
@jirevwe, I think we should abstract this into the `typsense` package so it can be adequately tested as part of the package. I am thinking of the right abstraction, something like `e.searcher.SearchIndex` that returns an array of ids using the same internal search method, I believe.

Reply:
What you described was the original behavior, I just changed it in this PR because the package assumed you index an "event" when in reality you should be able to index any other collection

Reply:
No, but the structure of each document is the same, right? They all have a `uid`. We can filter for ids without assuming their collection.